### PR TITLE
Graphql render velocity fields

### LINKED ISSUE #20407: Introduce ability to request velocity-rendered fields values in the GraphQL request
As a GraphQL Content API User I want to be able to specify in the GraphQL query if a given field should be returned velocity-rendered in the response.  

One way of achieving this would be by passing an GraphQL argument to the field indicating so. 

Example: 
```
{
  webPageContentCollection(query: "+identifier:911868a2-b1f5-475a-bd70-1def8a12f331") {
    title
    body(render: true)
    myTextArea(render: false)
    identifier
  }
}
```

### PR DESCRIPTION
**Pull request by dsilvam**
Adds a way to get velocity-renderable fields rendered in the response of a request to the GraphQL Content API. 

Sample request: both `body` and `myTextArea` have a "$date"
```
{
  webPageContentCollection(query: "+identifier:911868a2-b1f5-475a-bd70-1def8a12f331") {
    title
    body(render: true)
    myTextArea(render: false)
    identifier
  }
}
```

Sample response:
```
{
  "data": {
    "webPageContentCollection": [
      {
        "title": "contentWithVelocity",
        "body": "May 14, 2021 5:20:52 PM",
        "myTextArea": "$date",
        "identifier": "911868a2-b1f5-475a-bd70-1def8a12f331"
      }
    ]
  }
}
```

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/FieldDataFetcher.java [hunk 1]
```diff
@@ -48,10 +55,42 @@ public Object get(final DataFetchingEnvironment environment) throws Exception {
...
+            return renderField && isFieldRenderable(field)
+                    ? renderFieldValue(environment, fieldValue)
+                    : fieldValue;
         } catch (Exception e) {
             Logger.error(this, e.getMessage(), e);
             throw e;
         }
     }
+
+    private Object renderFieldValue(DataFetchingEnvironment environment, Object fieldValue) {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/FieldDataFetcher.java:70]: set to final

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/FieldDataFetcher.java [hunk 2]
```diff
@@ -48,10 +55,42 @@ public Object get(final DataFetchingEnvironment environment) throws Exception {
...
+        final String fieldValueAsStr = fieldValue.toString();
+
+        final HttpServletRequest request = ((DotGraphQLContext) environment.getContext())
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/FieldDataFetcher.java:73]: I prefer the DotGraphQLContext.class.cast(environment.getContext()). getHttpServletRequest()

But it is pretty much the same

**Review by jdotcms — APPROVED** [review]

**Comment by jdotcms** [comment]
May be a curl test?

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/FieldDataFetcher.java [hunk 3]
```diff
@@ -48,10 +55,42 @@ public Object get(final DataFetchingEnvironment environment) throws Exception {
...
+                .getHttpServletRequest();
+
+        final HttpServletResponse response = ((DotGraphQLContext) environment.getContext())
+                .getHttpServletResponse();
+
+        final org.apache.velocity.context.Context context = VelocityUtil
+                .getInstance().getContext(request, response);
+
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/FieldDataFetcher.java:81]: What else can/should we add to the velocity context here?    I think we should add at least the found contentlet object to the context as "content" and/or contentlet, like we do in the workflow processor

Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/graphql/datafetcher/FieldDataFetcher.java:81]: makes sense @wezell

**Comment by dsilvam** [comment]
@jdotcms postman done

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/FieldDataFetcher.java [hunk 4]
```diff
@@ -48,10 +43,47 @@ public Object get(final DataFetchingEnvironment environment) throws Exception {
                 }
             }
 
-            return fieldValue;
+            final boolean renderField = Try.of(()-> (boolean) environment.getArgument("render"))
```
Comment (victoralfaro-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/FieldDataFetcher.java:46]: I think this should not be called if `isRenderable(field)` returns `false`.

**Review by freddyucv — APPROVED** [review]

**Pull request merged**